python_modules/dagster/dagster/_record/__init__.py

```diff
@@ -358,16 +360,40 @@ def as_dict_for_new(obj) -> Mapping[str, Any]:
     return from_obj
 
 
-def copy(obj: TVal, **kwargs) -> TVal:
-    """Create a copy of this record instance, with new values specified as key word args."""
-    return obj.__class__(
+def copy(record: TVal, **kwargs) -> TVal:
+    """Create a new instance of this record type using its constructor with
+    the original records values overrode with new values specified as key args.
+    """
+    return record.__class__(
         **{
-            **as_dict_for_new(obj),
+            **as_dict_for_new(record),
             **kwargs,
         }
     )
 
 
+def replace(obj: TVal, **kwargs) -> TVal:
```

Comment:
open to better names for this as an alternative to `copy` , went with `replace` since thats the tuple method this behavior emulates